packages/pds/src/api/app/bsky/actor/getProfile.ts

```diff
@@ -10,93 +9,24 @@ export default function (server: Server, ctx: AppContext) {
     handler: async ({ auth, params }) => {
       const { actor } = params
       const requester = auth.credentials.did
+      const { db, services } = ctx
+      const actorService = services.actor(db)
 
-      const db = ctx.db.db
-      const { ref } = db.dynamic
+      const user = await actorService.getUser(actor, true)
 
-      const queryRes = await db
-        .selectFrom('did_handle')
-        .where(actorWhereClause(actor))
-        .innerJoin('repo_root', 'repo_root.did', 'did_handle.did')
-        .leftJoin('profile', 'profile.creator', 'did_handle.did')
-        .select([
-          'repo_root.takedownId',
-          'did_handle.did as did',
-          'did_handle.handle as handle',
-          'did_handle.actorType as actorType',
-          'did_handle.declarationCid as declarationCid',
-          'profile.uri as profileUri',
-          'profile.displayName as displayName',
-          'profile.description as description',
-          'profile.avatarCid as avatarCid',
-          'profile.bannerCid as bannerCid',
-          db
-            .selectFrom('follow')
-            .whereRef('creator', '=', ref('did_handle.did'))
-            .select(countAll.as('count'))
-            .as('followsCount'),
-          db
-            .selectFrom('follow')
-            .whereRef('subjectDid', '=', ref('did_handle.did'))
-            .select(countAll.as('count'))
-            .as('followersCount'),
-          db
-            .selectFrom('post')
-            .whereRef('creator', '=', ref('did_handle.did'))
-            .select(countAll.as('count'))
-            .as('postsCount'),
-          db
-            .selectFrom('follow')
-            .where('creator', '=', requester)
-            .whereRef('subjectDid', '=', ref('did_handle.did'))
-            .select('uri')
-            .as('requesterFollow'),
-          db
-            .selectFrom('mute')
-            .whereRef('did', '=', ref('did_handle.did'))
-            .where('mutedByDid', '=', requester)
-            .select('did')
-            .as('requesterMuted'),
-        ])
-        .executeTakeFirst()
-
-      if (!queryRes) {
-        throw new InvalidRequestError(`Profile not found`)
+      if (!user) {
+        throw new InvalidRequestError('Profile not found')
       }
-      if (softDeleted(queryRes)) {
+      if (softDeleted(user)) {
```

Comment:
actually looks like the service is handling this so you can probably ditch the extra check

Reply:
I think it does this so that it can provide the `AccountTakedown` error rather than a not-found— the `true` flag passed to service is letting it return taken-down accounts.  Good call, though, it may be smart to switch to just a not-found 🤔 I am going to merge this for now preserving the behavior and we can always tweak it.